UnityPy/math/Vector4.py

```diff
@@ -1,155 +1,112 @@
-        }
-
-        public bool Equals(Vector4 other)
-        {
-            return X.Equals(other.X) && Y.Equals(other.Y) && Z.Equals(other.Z) && W.Equals(other.W);
-        }
-
-        public void Normalize()
-        {
-            var length = Length();
-            if (length > kEpsilon)
-            {
-                var invNorm = 1.0f / length;
-                X *= invNorm;
-                Y *= invNorm;
-                Z *= invNorm;
-                W *= invNorm;
-            }
-            else
-            {
-                X = 0;
-                Y = 0;
-                Z = 0;
-                W = 0;
-            }
-        }
-
-        public float Length()
-        {
-            return (float)Math.Sqrt(LengthSquared());
-        }
-
-        public float LengthSquared()
-        {
-            return X * X + Y * Y + Z * Z + W * W;
-        }
-
-        public static Vector4 Zero => new Vector4();
-
-        public static Vector4 operator +(Vector4 a, Vector4 b)
-        {
-            return new Vector4(a.X + b.X, a.Y + b.Y, a.Z + b.Z, a.W + b.W);
-        }
-
-        public static Vector4 operator -(Vector4 a, Vector4 b)
-        {
-            return new Vector4(a.X - b.X, a.Y - b.Y, a.Z - b.Z, a.W - b.W);
-        }
-
-        public static Vector4 operator -(Vector4 a)
-        {
-            return new Vector4(-a.X, -a.Y, -a.Z, -a.W);
-        }
-
-        public static Vector4 operator *(Vector4 a, float d)
-        {
-            return new Vector4(a.X * d, a.Y * d, a.Z * d, a.W * d);
-        }
-
-        public static Vector4 operator *(float d, Vector4 a)
-        {
-            return new Vector4(a.X * d, a.Y * d, a.Z * d, a.W * d);
-        }
-
-        public static Vector4 operator /(Vector4 a, float d)
-        {
-            return new Vector4(a.X / d, a.Y / d, a.Z / d, a.W / d);
-        }
-
-        public static bool operator ==(Vector4 lhs, Vector4 rhs)
-        {
-            return (lhs - rhs).LengthSquared() < kEpsilon * kEpsilon;
-        }
-
-        public static bool operator !=(Vector4 lhs, Vector4 rhs)
-        {
-            return !(lhs == rhs);
-        }
-
-        public static implicit operator Vector2(Vector4 v)
-        {
-            return new Vector2(v.X, v.Y);
-        }
-
-        public static implicit operator Vector3(Vector4 v)
-        {
-            return new Vector3(v.X, v.Y, v.Z);
-        }
-
-        public static implicit operator Color(Vector4 v)
-        {
-            return new Color(v.X, v.Y, v.Z, v.W);
-        }
-
-        private const float kEpsilon = 0.00001F;
-    }
-}
-"""
+        if len(args) == 1:
+            args = args[0]
+
+        if isinstance(args, Sequence):
+            if len(args) == 4:  # args=(x, y, z, w)
+                self.X, self.Y, self.Z, self.W = args
+                return
+            if len(args) == 2:  # args=(Vector3, w)
+                self.X, self.Y, self.Z = args[0]
+                self.W = args[1]
```

Comment:
A return is missing here.
I guess that a simple 
if, elif, ..., ellif raise TypeError would be cleaner